conn.go

```diff
@@ -185,7 +185,6 @@ func (l *Conn) sendMessage(packet *ber.Packet) (chan *ber.Packet, error) {
 
 func (l *Conn) sendMessageWithFlags(packet *ber.Packet, flags sendMessageFlags) (chan *ber.Packet, error) {
 	if l.isClosing {
-		l.messageMutex.Unlock()
```

Comment:
I agree that this `Unlock()` is in the wrong place, but I think it should be before the return on line 198?